app/src/main/java/de/aurora/mggvertretungsplan/MainActivity.java

```diff
@@ -220,6 +221,68 @@ public boolean onOptionsItemSelected(MenuItem item) {
             case R.id.action_feedback:
                 launchCustomTabsIntent(color, getString(R.string.feedback_url));
                 break;
+            case R.id.action_share:
+                // Share
+                String shareText = "Vertretungsplan " + class_name;
+                String input = StorageUtilities.readFile(this);
+                try {
+                    JSONArray json = new JSONArray(input);
+                    // For every Week:
+                    for (int i = 0; i < json.length(); i++) {
+                        JSONObject dateData = json.getJSONObject(i);
+                        String Date = dateData.getString("date");
+                        String Week = dateData.getString("week");
+                        shareText = shareText + "\n" + Date + " Woche " + Week;
+                        JSONArray dateElements = dateData.getJSONArray("elements");
+                        // For every Element of the Week:
+                        for (int j = 0; j < dateElements.length(); j++){
+                            JSONObject actChange = dateElements.getJSONObject(j);
+                            String actClass = actChange.getString("class_name");
+                            String actHour = actChange.getString("hour");
+                            String actSubj = actChange.getString("subject");
+                            String act_newSubj = actChange.getString("newSubject");
+                            String actRoom = actChange.getString("room");
+                            String act_newRoom = actChange.getString("newRoom");
+                            String actInfo = actChange.getString("info");
+                            //Only if own Class:
+                            if (actClass.equals(class_name)) {
```

Comment:
This could lead to a bug, because classes can also be `10ab` and the saved class is only `10a`. I created the Grade.java file for that use case :) It makes classes comparable.

This is nothing you could know 🙈 - I just noticed it, because I had the same bug once in the past.